### PR TITLE
fix: disable notifications when basic functionality off

### PR DESCRIPTION
**Pull request by matteoscurati**
## **Description**

This PR ensures that notifications are correctly disabled when the user deactivates Basic functionality. The current implementation only disables profile sync when Basic functionality is turned off, leaving notifications in an incorrect state—they can’t be fetched but remain active. This causes a series of UI errors, the most obvious being that the counters stay visible.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28045?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. With notifications active
2. Deactivate Basic functionality
3. Verify that notifications cannot be enabled unless Basic functionality is reactivated
4. Verify that the counter no longer shows any unread notifications

### **Before**

Screenshot 2024-10-24 at 10 08 32

### **After**

Screenshot 2024-10-24 at 10 19 48

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/hooks/metamask-notifications/useProfileSyncing.ts [hunk 1]
```diff
@@ -83,6 +85,9 @@ export function useDisableProfileSyncing(): {
     try {
       // disable profile syncing
       await dispatch(disableProfileSyncingAction());
+
+      // list notifications to update the counter
+      await listNotifications();
```
Comment (matteoscurati) [inline ui/hooks/metamask-notifications/useProfileSyncing.ts:90]: Updating the list after the notifications have been disabled, the counter is set to zero

File: ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx [hunk 2]
```diff
@@ -74,15 +74,14 @@ export function NotificationsSettingsAllowNotifications({
   }, [isMetamaskNotificationsEnabled]);
 
   useEffect(() => {
-    if (isMetamaskNotificationsEnabled && !error) {
+    if (!error) {
       listNotifications();
     }
   }, [isMetamaskNotificationsEnabled, error, listNotifications]);
 
   const toggleNotifications = useCallback(async () => {
```
Comment (legobeat) [inline ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx:82]: Drive-by observation: This looks like it could be racey as it's not idempotent. Refactoring it from `toggleNotifications()` to `setNotificationsEnabled(boolean)` should make it more robust.

File: ui/hooks/metamask-notifications/useProfileSyncing.ts (hunk 1 above)
Comment (Prithpal-Sooriya) [reply ui/hooks/metamask-notifications/useProfileSyncing.ts:90]: nice, we can ticket this for a core change in future.

File: ui/pages/settings/security-tab/profile-sync-toggle/profile-sync-toggle.tsx [hunk 3]
```diff
@@ -31,14 +30,14 @@ import { getUseExternalServices } from '../../../../selectors';
 
 function ProfileSyncBasicFunctionalitySetting() {
   const basicFunctionality: boolean = useSelector(getUseExternalServices);
-  const { setIsProfileSyncingEnabled } = useSetIsProfileSyncingEnabled();
+  const { disableProfileSyncing } = useDisableProfileSyncing();
```
Comment (Prithpal-Sooriya) [inline ui/pages/settings/security-tab/profile-sync-toggle/profile-sync-toggle.tsx:33]: Nice, this hook will disable profile syncing and disable notifications.

FUTURE - we probs want to clean up the hooks.

File: ui/pages/settings/security-tab/profile-sync-toggle/profile-sync-toggle.tsx [hunk 4]
```diff
@@ -84,11 +82,11 @@ const ProfileSyncToggle = () => {
                 was_notifications_on: isMetamaskNotificationsEnabled,
               },
             });
+            disableProfileSyncing();
```
Comment (Prithpal-Sooriya) [inline ui/pages/settings/security-tab/profile-sync-toggle/profile-sync-toggle.tsx:85]: moving this after segment events so we can at least send to segment first.

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by mathieuartu — APPROVED** [review]
